.github/workflows/build.yml

```diff
@@ -1,24 +1,44 @@
-name: build
-on: push
+name: Build
+on:
+  push:
+    branches:
+      - main
 
 jobs:
   build:
-    runs-on: ubuntu-20.04
-    container:
-      image: openjdk:8-jdk
+    strategy:
+      matrix:
+        java: [
+          8, # Minimum
+          11,  # LTS
+          16,  # Latest
+        ]
+        os: [ubuntu-20.04]
```

Comment:
There's no reason to have this if the array only contains one entry. (Isn't java meant to be 'write once, run anywhere' anyways?)

Reply:
yeah, but this is just future proofing it, in case we ever do want to build on windows its already set up to do so.